crawl-ref/source/actor.cc

```diff
@@ -1032,6 +1033,107 @@ void actor::collide(coord_def newpos, const actor *agent, int pow)
     }
 }
 
+/**
+ * @brief Attempts to knock this actor away from a specific tile,
+ *        using repeated one-tile knockbacks. Any LOS requirements
+ *        must be checked by the calling function.
+ * @param cause The actor responsible for the knockback.
+ * @param source The tile this actor is being knocked directly away from.
+ * @param distance The number of single tile knockbacks attempted.
+ * @param strength Each attempt has probability size/strength of failing.
+ *                 If strength is 0, never fail.
+ * @param pow Cause 2d(pow/10 + 1) damage on collision.
```

Comment:
This comment is doomed to become obsolete. The caller does not need to know this math, and anyone who changes the behavior of collide() will never, ever remember to update this.

Instead, let's say something like `Determines damage done to us if we hit something.`